### ISSUE TITLE
Serializable class must have single primary constructor (iOS)

### ISSUE THREAD

**Issue by pauminku** [Kotlin/kotlinx.serialization#709]
**Describe the bug**
I'm using kotlin multiplatform for android and iOS in production apps without any problem until now that I added Serialization.
Serialization is working well on android, but fails on iOS compilation, with the following error: 

Compilation failed: Serializable class must have single primary constructor * Source files: [A list of all my multiplatform classes (not only @Serializable)] * Compiler version info: Konan: 1.3.61 / Kotlin: 1.3.60 * Output kind: LIBRARY |  
-- | --

**To Reproduce**
Serialization is added to serialize data classes like this one:
`@Serializable
data class Foo(val valueA: Int = 0, val valueB: Int = 0)`
I think this is all the relevant code, because is the main change that creates this unexpected behavior.

**Expected behavior**
I expected the same behavior on JVM and Native, in the sense that if it's working in Android I expect that works the same in iOS, or at least a more specific error message.

**Environment**
 - Kotlin version: 1.3.61
 - Library version: 0.14.0
 - Kotlin platforms: Native
 - Gradle version: 5.4.1
 - IDE version (think is not relevant) Android Studio 3.5.3

**Comment by sandwwraith** [comment]
Are all of your `@Serializable` classes `data`? This error is an internal assertion fail that should not happen

**Comment by pauminku** [comment]
Thanks for your quick response.
One class was not `data` because it has a constructor with parameters that are not properties, so I refactored a little my code to convert it to `data`class. It compiles and works fine. Thanks.

I played a little bit more and I think I've more details in what was happening. The problem wasn't the 'data' class, it was the absence of a  primary constructor in that class, as the error message says.

I think the word "single" in the error message confuse me, because primary constructor is unique by definition, so it makes me think the problem was on the secondary constructors.

The facts that compiles in android and the error message includes a list of all the classes instead of the problematic one didn't help to find the problem.

For reference, the changes to solve my problem,
insted of (works on Android but not in iOS):
```
@Serializable
class Foo  {
   
    internal constructor()

    constructor(paramA: Int = 0, paramB: Int = 0) {
    ...
    }
  }
```

Now I've (works in both, Android and iOS):
```
@Serializable
class Foo internal constructor() {
   
   constructor(paramA: Int = 0, paramB: Int = 0) : this() {
    ...
    }
  }
```

And works perfect. Thanks.
